src/ReactiveUI.Validation/Collections/ValidationText.cs

```diff
@@ -13,79 +16,237 @@ namespace ReactiveUI.Validation.Collections
+        public static ValidationText Create(params string[] validationTexts)
+        {
+            if (validationTexts is null || validationTexts.Length < 1)
+            {
+                return None;
+            }
+
+            // Optimise code path for single item array.
+            if (validationTexts.Length == 1)
+            {
+                var text = validationTexts[0];
+                if (text is null)
+                {
+                    return None;
+                }
+
+                if (text.Length < 1)
+                {
+                    return Empty;
+                }
+
+                return new ValidationText(validationTexts);
+            }
+
+            // Ensure we have no null items in the multi-item array
+            if (validationTexts.Any(t => t is null))
+            {
+                // Strip nulls
+                validationTexts = validationTexts.Where(t => t != null).ToArray();
+                if (validationTexts.Length < 1)
+                {
+                    return None;
+                }
+
+                if (validationTexts[0].Length < 1)
+                {
+                    return Empty;
+                }
+            }
+
+            return new ValidationText(validationTexts);
+        }
+
         /// <inheritdoc/>
         public IEnumerator<string> GetEnumerator()
         {
-            return _texts.GetEnumerator();
+            return ((IEnumerable<string>)_texts).GetEnumerator();
         }
 
         /// <inheritdoc/>
         IEnumerator IEnumerable.GetEnumerator()
         {
-            return GetEnumerator();
+            return _texts.GetEnumerator();
         }
 
         /// <summary>
         /// Adds a text to the collection.
         /// </summary>
         /// <param name="text">Text to be added in the collection.</param>
+        [ExcludeFromCodeCoverage]
+        [Obsolete("ValidationText will be made immutable in future versions, please do not use the Add(string) method.")]
         public void Add(string text)
         {
-            _texts.Add(text);
+            if (ReferenceEquals(this, Empty))
+            {
+                throw new InvalidOperationException("Adding to ValidationText.Empty is unsupported.");
+            }
+
+            if (ReferenceEquals(this, None))
+            {
+                throw new InvalidOperationException("Adding to ValidationText.None is unsupported.");
+            }
+
+            _texts = _texts.Concat(new[] { text }).ToArray();
         }
 
         /// <summary>
         /// Clear all texts.
         /// </summary>
+        [ExcludeFromCodeCoverage]
+        [Obsolete("ValidationText will be made immutable in future versions, please do not use the Clear() method.")]
         public void Clear()
         {
-            _texts.Clear();
+            if (ReferenceEquals(this, Empty))
+            {
+                throw new InvalidOperationException("Adding to ValidationText.Empty is unsupported.");
```

Comment:
Worth adjusting the error message to e.g. `Clearing ValidationText.Empty is unsupported` etc.